File: 07-selecao-enumeracoes-e-estruturas/exemplos/combustivel.cpp
#include "bscpp.hpp"

using namespace std;

// Representa o tipo do combustível.
enum Combustivel {
    Alcool,
    Gasolina,
};

// Indica o combustível que deve ser utilizado no abastecimento. Produz
// Alcool se preco_alcool for menor ou igual a 70% do preco_gasolina,
// produz Gasolina caso contrário.
Combustivel indica_combustivel(double preco_alcool, double preco_gasolina)
{
    Combustivel combustivel;
    if (preco_alcool <= 0.7 * preco_gasolina) {
        combustivel = Alcool;
    } else {
        combustivel = Gasolina;
    }
    return combustivel;
}

examples
{
    // Alcool
    // 4.000 <= 0.7 * 6.000 é true
    check_expect(indica_combustivel(4.000, 6.000), Alcool);
    // 3.500 <= 0.7 * 5.000 é true
    check_expect(indica_combustivel(3.500, 5.000), Alcool);
    // Gasolina
    // 4.000 <= 0.7 * 5.000 é false
    check_expect(indica_combustivel(4.000, 5.000), Gasolina);
}

int main()
{
    run_tests();
}

File: 07-selecao-enumeracoes-e-estruturas/exemplos/combustivel.py
def indica_combustivel(preco_alcool: float, preco_gasolina: float) -> str:
    '''
    Indica o combustível que deve ser utilizado no abastecimento. Produz
    'alcool' se *preco_alcool* for menor ou igual a 70% do *preco_gasolina*,
    caso contrário produz 'gasolina'.

    Exemplos
    >>> # 'alcool'
    >>> # preco_alcool <= 0.7 * preco_gasolina é True
    >>> indica_combustivel(4.00, 6.00)
    'alcool'
    >>> indica_combustivel(3.50, 5.00)
    'alcool'
    >>> # 'gasolina'
    >>> # preco_alcool <= 0.7 * preco_gasolina é False
    >>> indica_combustivel(4.00, 5.00)
    'gasolina'
    '''
    if preco_alcool <= 0.7 * preco_gasolina:
        combustivel = "alcool"
    else:
        combustivel = "gasolina"
    return combustivel
